update-format-crau/src/bin/test.rs

```diff
@@ -79,12 +83,91 @@ fn main() -> Result<(), Box<dyn Error>> {
         _ => None,
     };
 
-    let signatures = match signatures_bytes {
-        Some(ref bytes) => Some(proto::Signatures::parse_from_bytes(bytes)?),
-        None => None,
+    Ok(signatures_bytes.unwrap())
+}
+
+// parse_signature_data takes SigBuffer and public key file path.
+// Return only actual data, without version and special fields.
+fn parse_signature_data(sigbytes: &[u8], pubkeyfile: &str) -> Result<Box<[u8]>, Box<dyn Error>> {
+    // Signatures has a container of the fields, i.e. version, data, and
+    // special fields.
+    let sigmessage = proto::Signatures::parse_from_bytes(sigbytes)?;
+    let siglen = sigbytes.len();
```

Comment:
siglen is used below when calculating size of vec for sigbs.

I think we can simplify that part by writing `let mut sigbs: Box<[u8]> = vec![0; sigbytes.len()].into_boxed_slice();`, and removing siglen.

Reply:
ah, i think i see what you meant now!

the thing is, later in the func when you do
```rust
            Ok(sigxvec) => {
                sigbs = sigxvec.into_boxed_slice();
                sigfound = true;
                break;
            }
```

the `sigbs = ` line drops (frees) the boxed slice you allocate earlier in the function and replaces it with the `Vec<>` you get from `verify_sig_pubkey()` (after turning it into a boxed slice, of course). so, effectively, you're allocating a zero buffer just to free it later (rather than re-using the same buffer, which i think is what your intention was).

if you wanted to shorten the signature we get from `verify_sig_pubkey()`, you'll want to do that explicitly on the vec before you turn it into a boxed slice.